my-app/src/app/main-layout/main-layout.module.ts

```diff
@@ -20,6 +21,7 @@ import { LoginModule } from '../login/login.module';
     AddCourseComponent,
     BreadcrumbsComponent,
     CourseComponent,
+    CourseNewComponent,
```

Comment:
There should be a separate module for route 'courses'.
It should contain CoursesPageComponent, CourseListComponent, CourseListItemComponent and CourseEditorComponent. 
CourseEditorComponent is a better name for the component as far as it will be responsible not only for course creation but for editing as well

Reply:
Will work on this soon.